schema/home/home_page_hero_units.js

```diff
@@ -23,7 +53,32 @@ const HomePageHeroUnitType = new GraphQLObjectType({
     },
     title: {
       type: GraphQLString,
-      resolve: ({ mobile_title }) => mobile_title,
+      resolve: ({ mobile_title, name, platform }) => {
+        return platform === 'desktop' ? name : mobile_title;
+      },
+    },
+    title_image_url: {
+      args: {
+        retina: {
+          type: GraphQLBoolean,
+        },
+      },
+      type: GraphQLString,
+      resolve: ({ title_image_url, title_image_retina_url, retina }) => {
+        return retina ? title_image_retina_url : title_image_url;
+      },
+    },
+    subtitle: {
+      type: GraphQLString,
+      resolve: ({ mobile_description, description, platform }) => {
+        return platform === 'desktop' ? description : mobile_description;
```

Comment:
Since we're smoothing over the platform specific fields in other places like this I decided to go ahead and smooth over other data modeling parts including normalizing the free text fields of `type` and `menu_color_class` above into a more appropriate `mode` enum of the six states a hero unit can be in.